docs/source/Plugins.rst

```diff
@@ -157,6 +157,20 @@ plugin (``json``) will be executed before the XUnit plugin (``xunit``).
 If the HTML result plugin is installed and enabled (``html``) it will
 be executed before both JSON and XUnit.
 
+Configuring the plugin execution order
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+On some circumstances it may be necessary to change the order in which plugins
+are executed.  To do so, add a ``order`` entry a configuration file section
+named after the plugin type.  For ``job.prepost`` plugin types, the section name
+has to be named ``plugins.job.prepost``, and it would look like this::
+
+  [plugins.job.prepost]
+  order = ['myplugin', 'jobscripts']
```

Comment:
I'm also fine with the way it's done in this PR. Basically we have different hooks and we want to specify the order of the plugins executed in those points. If one desires to change the order, they can implement the plugin as a different type, but I don't see a reason to allow running `post` plugin before `pre` plugin, or `run` plugin before `results` plugin.

I have only one remark regarding usability, we might reconsider allowing command arguments to specify the order, because we might want to alter the order for remote execution, but keep it of non-remote execution.